### PR TITLE
move dataset out of experimental

### PR DESCRIPTION
**Pull request by grant-Kolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/dataset/evaluation.py [hunk 1]
```diff
@@ -26,31 +26,31 @@
...
+from kolena.dataset.dataset import _to_serialized_dataframe
+from kolena.dataset.dataset import load_dataset
 from kolena.errors import IncorrectUsageError
 from kolena.errors import NotFoundError
 
 TYPE_EVALUATION_CONFIG = Optional[Dict[str, Any]]
 TEST_ON_TYPE = Optional[Union[str, List[str]]]
 
 
-def _fetch_dataset(dataset: str) -> pd.DataFrame:
```
Comment (ly-kolena) [inline kolena/dataset/evaluation.py:53]: Not from this PR, but this doesn't look to be used?

Comment (grant-Kolena) [reply kolena/dataset/evaluation.py:53]: this is used in an unused function in [shipyard](https://github.com/kolenaIO/shipyard/blob/5bc7515ac9617f4fb0b0f6fd1f9090ecad062bab/galleon/galleon/services/v1/ingest/lib/data_ingestion/_utils/dataset.py#L36), I think having the ability to fetch dataset on demand by name is useful in SDK, I used it in my binary classification example  [PR](https://github.com/kolenaIO/kolena/pull/377/files)

Comment (ly-kolena) [reply kolena/dataset/evaluation.py:53]: Can this be covered by `dataset.py::fetch_dataset`?

Comment (grant-Kolena) [reply kolena/dataset/evaluation.py:53]: oh I see, this is a dup then, I will remove this

**Review by ly-kolena — APPROVED** [review]

**Pull request merged**